### ISSUE TITLE
Audio spatialization improvements

### ISSUE THREAD

**Issue by trethaller** [HeapsIO/heaps#651]
Attaching sounds to moving objects pretty standard. Current API doesn't enable this out of the box, the `position`, `velocity` and `direction` properties of `hxd.snd.effect.Spatialization` have to be updated manually by the game code.

Possible improvements:

- Replacing `position` and `direction` by a `h3d.Matrix`, which would allow hooking-up to a `h3d.scene.Object.absPos` and get position updates for free. This doesn't solve the problem of recomputing velocity
- Adding a `followObject` property, and add an `update()` method to `hxd.snd.Effect`, so we can update the fields in `Spatialization`. There we could also calculate the velocity

Other considerations:

- Currently there isn't an easy way to position 2D sounds. Attaching sounds to 2D objects is also a very common use case 

Looking for feedback and/or volunteers

**Comment by Yanrishatum** [comment]
How 2D spatialization would work however? Just panning tweaks?

**Comment by trethaller** [comment]
Yes basically, right now you have to put a listener at a fixed distance and move objects on a plane to simulate that

**Comment by jefvel** [comment]
I wanted to look into this, but am currently having problems with getting the spatialization to work.

It seems to do nothing on both mac and windows. The volume of clips seem to fade out and in, but that is purely based on the `fadeDistance` parameter. The parameters that are set in openAL seem to do nothing. The spatialization sliders in the sound sample doesn't change the sound.
